api/v3/common_types.go

```diff
@@ -104,8 +104,11 @@ type CommonSplunkSpec struct {
 	// Full path or URL for a Splunk Enterprise license file
 	LicenseURL string `json:"licenseUrl"`
 
-	// LicenseMasterRef refers to a Splunk Enterprise license manager managed by the operator within Kubernetes
-	LicenseMasterRef corev1.ObjectReference `json:"licenseMasterRef"`
+	// ObsoleteLicenseManagerRef refers to a Splunk Enterprise license manager managed by the operator within Kubernetes
+	ObsoleteLicenseManagerRef corev1.ObjectReference `json:"licenseMasterRef"`
```

Comment:
The json tag(licenseMasterRef) is different from the field name(ObsoleteLicenseManagerRef). 

Two cents on this: 

IMO the intention here is to:

1.  Keep the conf exposed to the customer for LM ref the same i.e `licenseMasterRef`
2. Have the code refer the same with an `Obsolete*` tag.

Naming it `ObsoleteLicenseManager` seems a bit confusing to me. Maybe we can name it `ObsoleteLicenseMaster` or `LicenseMaster` thus making it:
- Similar to the json tag 
- Making the names unique enough for us to come back and remove in the next phase when we don't support the older LM.
- If using `ObsoleteLicenseMaster` spells out explicitly that it is obsolete.

Reply:
This is because we need to rename it without using the Bias language `Master`.  

It's a bit confusing because we can't change the Ref as long as the LicenseMaster CRD is supported, but at the same time using `Master` for variable names breaks the BiasLang effort. 

I choose `Obsolete*` as a term to mark every code that can be removed once the support for the LicenseMaster is deprecated. But we can discuss/think about other naming options as well.

Reply:
One thing I wanted to check on is whether the bias lang efforts are considering the json tags as well i.e LicenseMasterRef ? If yes, then we could still use ObsoleteLicenseMaster everywhere since the bias lang efforts would be broken either way? It should be fine either way - just an implementation choice :)

Reply:
Yes, the effort considers json tags, so that's true it will only be completed once they are all removed. Maybe in 3.0 release? :D

We avoided changing the json tags because of the impact on existing deployments. Also, the reference is used more often than the tag itself, so we picked the one with more impact.

Reply:
@akondur just FYI we're rolling back the Obsolete naming to keep the original name LicenseMasterRef.

Reply:
Great, thanks for letting me know @mgaldino-splunk